### DISCUSSION TITLE
exec --no-startup-id i3-msg 'workspace $ws1; exec alacritty' opens alacritty in $ws2

### DISCUSSION THREAD

**Discussion by adiSuper94** [i3/i3#5525] [Q&A]
exec --no-startup-id i3-msg 'workspace $ws1; exec alacritty' opens alacritty in $ws2.
And if if move my cursor quickly during startup to a diff workspace it opens there.

Facing this issue only after upgrading from 4.19/4.20  to 4.22.
OS: Ubuntu 23.04

**Comment by martisj** [comment]
What does your config look like?

**Reply by adiSuper94** [reply]
https://github.com/adiSuper94/config/blob/72435d035e2988af59b8ecd9ba81f98961756982/i3/config#L212

It is commented out right now cuz I hate having move it around on every restart.

**Comment by okraits** [comment]
You should rather use _assign_: https://i3wm.org/docs/userguide.html#assign_workspace

**Reply by adiSuper94** [reply]
I do not want every new instance of an application to open up in a specific workspace.
Especially if the application is the terminal.

I do use assign for for some other application.

**Reply by okraits** [reply]
Nevertheless, what you mentioned is prone to race conditions. You could improve it by using `sleep` in between but there can always be issues when using something like that.

**Reply by orestisfl** [reply]
Specifically about alacritty, you can start it with the `--class` flag and assign new rules for that class

**Comment by vincentbernat** [comment]
Alacritty does not support the startup protocol that would allow i3 to put the window in the right workspace on start. Author does not intend to implement it: https://github.com/alacritty/alacritty/issues/2824.

**Reply by adiSuper94** [reply]
I am just wondering why it worked before the upgrade though 🤔
